src/common/k8s-api/__tests__/kube-api.test.ts

```diff
@@ -409,21 +423,220 @@ describe("KubeApi", () => {
           });
         });
       });
+
+      describe("when resource request fufills with no resource", () => {
+        beforeEach(async () => {
+          await fetchMock.resolveSpecific(
+            ["http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1"],
+            createMockResponseFromString("http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1", JSON.stringify({
+              resources: [],
+            })),
+          );
+        });
+
+        it("requests resources from the second versioned api group from the initial apiBase", () => {
+          expect(fetchMock.mock.lastCall).toMatchObject([
+            "http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1",
+            {
+              headers: {
+                "content-type": "application/json",
+              },
+              method: "get",
+            },
+          ]);
+        });
+
+
+
+        describe("when resource request fufills with a resource", () => {
+          beforeEach(async () => {
+            await fetchMock.resolveSpecific(
+              ["http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1"],
+              createMockResponseFromString("http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1", JSON.stringify({
+                resources: [{
+                  name: "ingresses",
+                }],
+              })),
+            );
+          });
+
+          it("makes the request to get the resource", () => {
+            expect(fetchMock.mock.lastCall).toMatchObject([
+              "http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1/namespaces/default/ingresses/foo",
+              {
+                headers: {
+                  "content-type": "application/json",
+                },
+                method: "get",
+              },
+            ]);
+          });
+
+          it("sets fields in the api instance", () => {
+            expect(ingressApi).toEqual(expect.objectContaining({
```

Comment:
Mutating fields of an "API" is a pretty shady practice, if you can agree without my effort to justify :) Let's try thinking of alternate patterns elsewhere in the future.

Reply:
oh sure, this is just part of the existing API